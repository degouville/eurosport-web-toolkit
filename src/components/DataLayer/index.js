import React from 'react';
import PropTypes from 'prop-types';

class DataLayer extends React.Component {
  componentDidMount() {
    const { dataLayer } = this.props;
    if (dataLayer) {
      this.dataLayerScript = document.createElement('script');
      this.dataLayerScript.innerHTML = `window.dataLayer = window.dataLayer || []; window.dataLayer.push(${JSON.stringify(
        dataLayer
      )});`;
      document.body.appendChild(this.dataLayerScript);
    }
  }

  componentWillUnmount() {
    if (this.dataLayerScript) {
      document.body.removeChild(this.dataLayerScript);
    }
  }

  render() {
    return null;
  }
}

DataLayer.defaultProps = {
  dataLayer: null,
};

DataLayer.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  dataLayer: PropTypes.object,
};

export default DataLayer;
