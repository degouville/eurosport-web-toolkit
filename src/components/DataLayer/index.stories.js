/* eslint-disable no-underscore-dangle */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { object } from '@storybook/addon-knobs';
import { DataLayer } from '../..';
import dataLayerMock from './mocks/dataLayerMock';

const datalayerStories = storiesOf('Components|DataLayer', module).addDecorator(withInfo);

datalayerStories.add('configurable', () => (
  <DataLayer dataLayer={object('dataLayer', dataLayerMock.dataLayer, 'dataLayer prop example')} />
));
