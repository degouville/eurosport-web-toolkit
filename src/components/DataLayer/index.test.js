import React from 'react';
import { render, mount, shallow } from 'enzyme';
import DataLayer from '.';
import dataLayerMock from './mocks/dataLayerMock';

describe('<DataLayer />', () => {
  it('renders with expected snapshot', () => {
    expect(render(<DataLayer dataLayer={dataLayerMock.dataLayer} />)).toMatchSnapshot();
  });

  it('renders with expected dataLayer props', () => {
    // eslint-disable-next-line no-unused-vars
    const wrapper = mount(<DataLayer dataLayer={dataLayerMock.dataLayer} />);
    expect(window.dataLayer[0]).toMatchObject(dataLayerMock.dataLayer);
  });

  it('should not render when no datalayer is passed', () => {
    const wrapper = shallow(<DataLayer />);
    expect(wrapper.find('DataLayer').exists()).toBeFalsy();
  });
});
