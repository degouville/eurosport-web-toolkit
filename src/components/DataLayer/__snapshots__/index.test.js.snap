// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataLayer /> renders with expected snapshot 1`] = `null`;
